addons/mass_mailing/models/mailing.py

```diff
@@ -467,18 +467,23 @@ def _get_ab_testing_description_modifying_fields(self):
     def create(self, vals_list):
         ab_testing_cron = self.env.ref('mass_mailing.ir_cron_mass_mailing_ab_testing').sudo()
         for values in vals_list:
-            if values.get('body_html'):
-                values['body_html'] = self._convert_inline_images_to_urls(values['body_html'])
             if values.get('ab_testing_schedule_datetime'):
                 at = fields.Datetime.from_string(values['ab_testing_schedule_datetime'])
                 ab_testing_cron._trigger(at=at)
         mailings = super().create(vals_list)
         mailings._create_ab_testing_utm_campaigns()
         mailings._fix_attachment_ownership()
 
+        for values, mailing in zip(vals_list, mailings):
+            if values.get('body_arch'):
+                mailing.body_arch = mailing._convert_inline_images_to_urls(mailing.body_arch)
```

Comment:
Is that what we want? I'm guessing it is but it's a rare case of something from `body_arch` changing after you save it.

Reply:
The goal of calling `convert_inline_images_to_urls` *after* `super().create` is to have the record's id when creating attachments (so they can be properly linked to the mailing record via the res_id field).
Maybe I should pop out `body_arch` and `body_html` from the `values` dictionary before `create`, and then set their values after record creation and image convertion... ? (The end result would be the same, but it would make more sense to only set those fields once...)

Reply:
What I meant was that this line will transform the html that is saved in `body_arch` whereas usually we only transform `body_html` and leave `body_arch` as it was.

Reply:
Yes, indeed. That's the whole point of this commit: https://github.com/odoo/odoo/pull/138563/commits/8664d177fe881b26838060d3a5c35347bd061813

> Before this commit, base64-encoded images were converted to attachments
only for the body_html field, leaving the body_arch's content unchanged.
As a result, every time a record was updated, the body_html content,
which is generated from the body_arch's content, would have the same
base64-encoded images, which were converted again to new attachments.
> ...
>This commit makes sure the body_arch has its images converted to
attachments as well, and also makes sure such attachments are reused
when replacing the image sources in the body_html, instead of creating
new ones.

Reply:
Oops sorry, got it. All good then :-D

Reply:
I'm probably missing something but I don't really get why converting body_arch is necessary, still
If we reuse the images anyway do we really need to convert the arch first?

Reply:
You're right that converting the body_arch is not necessary to prevent the creation of multiple attachments for the same image.
But, what I failed to mention in the commit message is that storing html fields containing base64-encoded images in the database is not desirable, mainly because of its size. In a way, the present task/PR is a follow-up of this one: https://www.odoo.com/web#id=3289061&cids=1&menu_id=4720&action=333&active_id=1695&model=project.task&view_type=form / https://github.com/odoo/odoo/pull/128452

Reply:
I see, that makes sense thanks.
Seems good to me then :)